### PR TITLE
Align AWS implementation with controller based implementation

### PR DESCRIPTION
**Pull request by prashanth26**
**What this PR does / why we need it**:
Aligning with controller based implmentation - https://github.com/gardener/machine-controller-manager/pull/460. 

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```noteworthy operator
Align AWS implementation with controller based implementation
```

### CODE REVIEW

File: pkg/aws/apis/aws_provider_spec.go [hunk 1]
```diff
@@ -13,7 +13,7 @@ limitations under the License.
 
 package api
 
-//AWSProviderSpec is the spec to be used while parsing the gRPC calls.
+//AWSProviderSpec is the spec to be used while parsing the calls.
```
Comment (ggaurav10) [inline pkg/aws/apis/aws_provider_spec.go:16]: are we using `type Secrets struct` defined in this file anymore? If not, can we clean it up?

File: pkg/aws/aws.go [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+	}
+
+	requiredInstance := instances[0]
+
+	response := &driver.GetMachineStatusResponse{
+		NodeName:   *requiredInstance.PrivateDnsName,
+		ProviderID: encodeProviderID(providerSpec.Region, *requiredInstance.InstanceId),
+	}
+
+	klog.V(2).Infof("Machine get request has been processed successfully for %q", req.Machine.Name)
```
[the remark this answers is not in the document]
Comment (prashanth26) [reply pkg/aws/aws.go]: Good point. I have decreased the log level to 1. I hope that is better.

File: pkg/aws/aws.go [hunk 3]
```diff
@@ -0,0 +1,366 @@
...
+	response := &driver.CreateMachineResponse{
+		ProviderID: encodeProviderID(providerSpec.Region, *runResult.Instances[0].InstanceId),
+		NodeName:   *runResult.Instances[0].PrivateDnsName,
+	}
+
+	klog.V(2).Infof("VM with Provider-ID: %q created for Machine: %q", response.ProviderID, machine.Name)
+	return response, nil
+}
+
+// DeleteMachine deletes the fake machine
```
[the remark this answers is not in the document]
Comment (prashanth26) [reply pkg/aws/aws.go]: Updated the comments

File: pkg/aws/apis/aws_provider_spec.go (hunk 1 above)
Comment (prashanth26) [reply pkg/aws/apis/aws_provider_spec.go:16]: Yes nice catch. Have removed them with this commit - https://github.com/gardener/machine-controller-manager-provider-aws/pull/10/commits/de39f3f87c093d4f442677de5282f0412fce224b

**Comment by prashanth26** [comment]
@ggaurav10 
> some minor suggestions.

All your comments and suggestions have been valuable and have resolved them here - PTAL - https://github.com/gardener/machine-controller-manager-provider-aws/pull/10/commits/de39f3f87c093d4f442677de5282f0412fce224b.

**Comment by hardikdr** [comment]
@prashanth26 Any plans of this merging this any soon?
- I see external contributors getting confused as this repo still talks about gRPC. I suggest we change the description of the repo to indicate it's `Deprecated` or not to be used, if this PR can't be merged soon.

**Comment by prashanth26** [comment]
Hi @hardikdr , Hi @ggaurav10 

PTAL at the changes, if it looks good we can merge them once and fix any issues post that. Will create a separate PR for the migration controller changes.

**Pull request merged**